instrumentation/rack/lib/opentelemetry/instrumentation/rack/middlewares/tracer_middleware.rb

```diff
@@ -15,6 +15,8 @@ module Middlewares
         # TracerMiddleware propagates context and instruments Rack requests
         # by way of its middleware system
         class TracerMiddleware # rubocop:disable Metrics/ClassLength
+          THREAD_KEY = :__opentelemetry_rack_span__
```

Comment:
Why not use `OpenTelemetry::Context` here instead of explicit thread locals? It'd look something like https://github.com/open-telemetry/opentelemetry-ruby/blob/7bf30855fa5353701457ecac183fc495886cbec3/api/lib/opentelemetry/trace.rb#L14-L16 and https://github.com/open-telemetry/opentelemetry-ruby/blob/7bf30855fa5353701457ecac183fc495886cbec3/api/lib/opentelemetry/trace.rb#L46-L73
probably on the `OpenTelemetry::Instrumentation::Rack` module.

Reply:
This is much better 😄